awscli/customizations/wizard/app.py

```diff
@@ -49,14 +52,15 @@ class InvalidChoiceException(Exception):
 
 
 class WizardTraverser:
-    def __init__(self, definition, values):
+    def __init__(self, definition, values, api_invoker):
         self._definition = definition
         self._values = values
-        self._prompt_definitions = self._collect_all_prompt_definitions()
+        self._prompt_definitions = self._collect_prompt_definitions()
         self._prompt_to_sections = self._map_prompts_to_sections()
         self._current_prompt = self._get_starting_prompt()
         self._previous_prompts = []
         self._visited_sections = [self.get_current_section()]
+        self._api_call_step = core.APICallStep(api_invoker=api_invoker)
```

Comment:
I don't think we need to include the `api_call_step` in the initializer any more. We'll probably have to clean this up in the tests as well.